proxy/config.go

```diff
@@ -0,0 +1,79 @@
+package config
+
+import (
+	"io/ioutil"
+
+	"gopkg.in/yaml.v2"
+)
+
+type AMQPConfig struct {
+	URL      string `yaml:"url,omitempty"`
+	Exchange string `yaml:"exchange,omitempty"`
+}
+
+// TLSConfig configures the options for TLS connections.
+type TLSConfig struct {
+	// The CA cert to use for the targets.
+	CAFile string `yaml:"ca_file,omitempty"`
+	// The client cert file for the targets.
+	CertFile string `yaml:"cert_file,omitempty"`
+	// The client key file for the targets.
+	KeyFile string `yaml:"key_file,omitempty"`
+	// Used to verify the hostname for the targets.
+	ServerName string `yaml:"server_name,omitempty"`
+	// Disable target certificate validation.
+	InsecureSkipVerify bool `yaml:"insecure_skip_verify"`
+
+	// Catches all undefined fields and must be empty after parsing.
+	XXX map[string]interface{} `yaml:",inline"`
```

Comment:
It's not go, it's an external library for parsing yaml.
This is here to make sure that our config file doesnt have unnecessary (illegal) definitions. I don't care about it myself, and this struct was copied from prometheus.